src/txn/txn_rollback_to_stable.c

```diff
@@ -0,0 +1,394 @@
+/*-
+ * Copyright (c) 2014-2017 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#ifdef HAVE_TIMESTAMPS
+
+/*
+ * __txn_rollback_to_stable_check --
+ *	Ensure the rollback request is reasonable.
+ */
+static int
+__txn_rollback_to_stable_check(
```

Comment:
I'd prefer this be moved down closer to its only caller where the API function is rather than all the way up here in the file.